packages/medusa/src/strategies/product/import.ts

```diff
@@ -0,0 +1,772 @@
+ * @param regex - A regex used to pick which keys are going to be copied in the new object
+ */
+function pickObjectPropsByRegex(
+  data: TParsedRowData,
+  regex: RegExp
+): TParsedRowData {
+  const variantKeyPredicate = (key: string): boolean => regex.test(key)
+  const ret = {}
+
+  for (const k in data) {
+    if (variantKeyPredicate(k)) {
+      ret[k] = data[k]
+    }
+  }
+
+  return ret
+}
+
+/**
+ * Pick data from parsed CSV object relevant for product create/update and remove prefixes from keys.
+ */
+function transformProductData(data: TParsedRowData): TParsedRowData {
+  const ret = {}
+  const productData = pickObjectPropsByRegex(data, /product\./)
+
+  Object.keys(productData).forEach((k) => {
+    const key = k.split("product.")[1]
+    ret[key] = productData[k]
+  })
+
+  return ret
+}
+
+/**
+ * Pick data from parsed CSV object relevant for variant create/update and remove prefixes from keys.
+ */
+function transformVariantData(data: TParsedRowData): TParsedRowData {
+  const ret = {}
+  const productData = pickObjectPropsByRegex(data, /variant\./)
+
+  Object.keys(productData).forEach((k) => {
+    const key = k.split("variant.")[1]
+    ret[key] = productData[k]
+  })
+
+  // include product handle to keep track of associated product
+  ret["product.handle"] = data["product.handle"]
+  ret["product.options"] = data["product.options"]
+
+  return ret
+}
+
+/**
+ * Default strategy class used for a batch import of products/variants.
+ */
+class ProductImportStrategy extends AbstractBatchJobStrategy<ProductImportStrategy> {
+  static identifier = "product-import"
+
+  static batchType = "product_import"
+
+  private processedCounter = 0
+
+  protected readonly redisClient_: IORedis.Redis
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly fileService_: IFileService<typeof FileService>
+
+  protected readonly productService_: ProductService
+  protected readonly batchJobService_: BatchJobService
+  protected readonly productVariantService_: ProductVariantService
+  protected readonly shippingProfileService_: typeof ShippingProfileService
+
+  protected readonly regionRepo_: typeof RegionRepository
+  protected readonly productRepo_: typeof ProductRepository
+  protected readonly productOptionRepo_: typeof ProductOptionRepository
+  protected readonly productVariantRepo_: typeof ProductVariantRepository
+
+  protected readonly csvParser_: CsvParser<
+    ProductImportCsvSchema,
+    Record<string, string>,
+    Record<string, string>
+  >
+
+  constructor(container) {
+    super(container)
+
+    const {
+      batchJobService,
+      productService,
+      productRepository,
+      productOptionRepository,
+      productVariantService,
+      productVariantRepository,
+      shippingProfileService,
+      regionRepository,
+      fileService,
+      redisClient,
+      manager,
+    } = container
+
+    this.csvParser_ = new CsvParser(container, CSVSchema)
```

Comment:
suggestion: The CsvParser does not use the container at all. should we remove that dependence?